caf-logging-logback-tomcat/src/main/java/com/github/cafapi/logging/logback/tomcat/CAFAccessLogConsoleAppender.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019-2020 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.logging.logback.tomcat;
+
+import ch.qos.logback.access.spi.AccessEvent;
+import ch.qos.logback.core.ConsoleAppender;
+
+import com.github.cafapi.logging.common.ProcessAndThreadIdProvider;
+
+public final class CAFAccessLogConsoleAppender extends ConsoleAppender<AccessEvent>
+{
+    public CAFAccessLogConsoleAppender()
+    {
+    }
+
+    @Override
+    protected void subAppend(final AccessEvent event)
+    {
+        final StringBuilder logMsgBuilder = new StringBuilder();
+        logMsgBuilder.append(ProcessAndThreadIdProvider.getId());
+        logMsgBuilder.append(" ");
+        if(event.getStatusCode() >= 400 && event.getStatusCode() <=499){
+            logMsgBuilder.append("WARN ");
+        }else if(event.getStatusCode() >= 500 && event.getStatusCode() <=599){
+            logMsgBuilder.append("ERROR");
+        }else{
+            logMsgBuilder.append("INFO ");
+        }
+        event.setThreadName(logMsgBuilder.toString());
```

Comment:
Your highjacking the threadname for the log level, instead investigate a custom layout implementation

Reply:
A new converter added for Loglevel and a new encoder created to use the new converter.